stylesheets/_component.buttons.scss

```diff
@@ -1,385 +1,110 @@
+    .icon,
+    [class*=icon-] {
+        font-size: 14px;
     }
 
-    &:focus {
+    &:focus,
+    &:hover:focus {
         @include pulsar-button-focused;
     }
 
-    &:hover,
-    &.is-hovered {
-        transform: translateY(1px);
-
-        // Uses a pseudo-element to fill the gap that caused the jitter
-        &::after {
-            bottom: 0;
-            content: '';
-            height: calc(100% + 5px);
-            left: -.05%; // make it slightly wider to prevent side jitter
-            position: absolute;
-            top: -3px;
-            width: 101%; // make it slightly wider to prevent side jitter
-            z-index: -1;
-        }
-
-        &:focus {
-            box-shadow: 0 1px 0 color(black);
-        }
-
-        &:focus:active {
-            box-shadow: none;
-        }
-    }
-}
-
-/**
- * Mixin to theme button states
- */
-
-@mixin button-states($state) {
-    $shadow-color: darken(color($state), 10);
-
-    background-color: color($state);
-    box-shadow: 0 2px 0 $shadow-color;
-    color: color($state, alt) !important; // should be immutable
-    position: relative;
-
-    &:hover,
-    &.is-hovered {
-        box-shadow: 0 1px 0 $shadow-color;
-        color: color($state, alt) !important;  // should be immutable
-    }
-
-    &:focus {
-        @include pulsar-button-focused;
-    }
-
-    &:active,
-    &.active {
-        background-color: darken(color($state), 5);
-        box-shadow: 0 0 0 $shadow-color;
-        color: color($state, alt) !important;
-        top: 2px;
-    }
-
     &:disabled,
-    &.is-disabled,
-    &.disabled {
+    &.disabled,
+    &.is-disabled {
         opacity: .5;
-
-        &.btn--naked {
-            color: color($state, dark) !important;
-
-            &:active,
-            &.active {
-                background-color: transparent !important;
-                color: color($state, dark) !important;
-            }
-        }
-    }
-
-    > .badge {
-        background-color: color($state, dark);
-        color: pick_best_color(color($state, dark), (#fff, color(gray, dark))) !important;
-
-        &.badge--outline {
-            background-color: transparent;
-            color: inherit !important;
-        }
-    }
-
-    &.btn--naked {
-        color: color($state, dark) !important;
-
-        &:active,
-        &.active {
-            background-color: color($state, light) !important;
-            color: color($state, dark) !important;
-
-            &:not(:disabled):not(.is-disabled).btn--#{$btn-white} {
-                color: color(text) !important;
-            }
-        }
-    }
-}
-
-/**
- * UI states
- */
-
-.btn {
-    @include button-states('base');
-    border-bottom-style: solid;
-    text-decoration: none;
-
-    &:disabled,
-    &.is-disabled,
-    &.disabled {
-        cursor: not-allowed !important;
+        pointer-events: none;
     }
 }
 
+// State variations
 @each $state, $state-color, $state-color-alt in $state-colors {
     .btn--#{'' + $state} {
```

Comment:
Wrap in `@if $state != 'base' {}` to avoid unused `--base` styles being outputted in compiled CSS.